pkg/controller/selector/filter.go

```diff
@@ -15,191 +15,304 @@
 package selector
 
 import (
+	"encoding/json"
 	"errors"
+	"regexp"
+	"strconv"
+	"strings"
 
+	log "github.com/golang/glog"
 	"github.com/opensds/opensds/pkg/model"
 )
 
-type Filter interface {
-	Handle(request map[string]interface{}, pools []*model.StoragePoolSpec) ([]*model.StoragePoolSpec, error)
+// StructToMap ...
+func StructToMap(structObj interface{}) (map[string]interface{}, error) {
```

Comment:
IMO, it's a public method which could be used by other packages. So it could be better if you move this method to `utils/utils.go`, or you can create a `utils` package under selector.